tests/crypto_test.py

```diff
@@ -388,6 +389,49 @@ def test_luks_get_uuid(self):
     def test_luks2_get_uuid(self):
         self._get_uuid(self._luks2_format)
 
+class CryptoTestGetMetadataSize(CryptoTestCase):
+
+    @unittest.skipIf("SKIP_SLOW" in os.environ, "skipping slow tests")
+    @unittest.skipUnless(HAVE_LUKS2, "LUKS 2 not supported")
+    def test_luks2_get_metadata_size(self):
+        """Verify that getting LUKS 2 device metadata size works"""
+
+        succ = self._luks2_format(self.loop_dev, PASSWD, None)
+        self.assertTrue(succ)
+
+        meta_size = BlockDev.crypto_luks_get_metadata_size(self.loop_dev)
+
+        ret, out, err = run_command("cryptsetup luksDump %s" % self.loop_dev)
+        if ret != 0:
+            self.fail("Failed to get LUKS 2 header from %s:\n%s %s" % (self.loop_dev, out, err))
+
+        m = re.search(r"offset:\s*([0-9]+)\s*\[bytes\]", out)
+        if m is None:
+            self.fail("Failed to get LUKS 2 offset information from %s:\n%s %s" % (self.loop_dev, out, err))
+        offset = m.group(1)
+        self.assertEquals(str(meta_size), offset, "LUKS 2 metadata sizes differ")
```

Comment:
I think it would be better to compare integers here too (just like in the other test below).

Reply:
Fixed.